### PR TITLE
Document 3D navigation for Godot 4

### PR DESCRIPTION
**Pull request by skyace65**
Documents how 3D navigation works in Godot 4, I should have gotten this done ages ago. Once the docs have a 3.5 branch I can make a pull request to add this to the 3.5 docs in addition to information on the old navigation method. I took out the description of the navigation mesh from this since that's not currently showing up in the latest build of 4.0, might just be a bug, I'll add that back in if needed. The one image this PR adds has gone through image compression.

### CODE REVIEW

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Looks good on a first read, will wait a bit for others to give it a look and also to potentially give it another read later. Thanks and good work!

**Comment by hilfazer** [comment]
Found a typo: `NavigationObstancle3D`

Will you do docs for 2D Navigation as well?

**Comment by skyace65** [comment]
@hilfazer I've fixed that typo. I'll try to document 2D navigation in the future but not as part of this PR, I haven't played around with it yet.

**Comment by AndreaCatania** [comment]
Thx a lot for this doc, it's really needed; Just few adjustments and it should be good to go!!

File: tutorials/navigation/real_time_navigation_3d.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+agents.
+
+To create a navigation region add the `NavigationRegion3D<class_NavigationRegion3D>`
+node to a 3D scene. Next create or add a :ref:`NavigationMesh<class_NavigationMesh>`
+in the inspector. The navmesh contains options for how it will be generated
+when it's baked. A full description of each setting and how it works can be
+found in the :ref:`NavigationMesh class reference<class_NavigationMesh>`.
+
+Once the settings have been properly configured press the "Bake NavMesh"
+button at the top of the inspector to generate it.
```
[the remark this answers is not in the document]
Comment (skyace65) [reply tutorials/navigation/real_time_navigation_3d.rst:32]: I'm not sure what you mean. I don't see any bounding boxes to adjust with a navigation region.

**Comment by skyace65** [comment]
I've made the suggested changes except for the bounding box. I've also split the navigation agent section in half and made a section on getting RVO to work.

File: tutorials/navigation/real_time_navigation_3d.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndreaCatania) [reply tutorials/navigation/real_time_navigation_3d.rst:32]: Right my bad. No bounding boxes though you need to set all the meshes under the navigation region. Do you think we should add this detail?

**Comment by AndreaCatania** [comment]
@skyace65 It looks nice, thanks!!
Feel free to address this: https://github.com/godotengine/godot-docs/pull/5690#discussion_r849088710 otherwise can be merged

**Review by AndreaCatania — APPROVED** [review]

**Comment by skyace65** [comment]
I added a sentence about geometry settings right after the navmesh is created. I also added the section from the 3.x version of this page going over the transparent navigation mesh you see after baking. It wasn't removed like I thought, I'm just stupid and set the minimum size for the navigation mesh too low. If both of these changes are good I'm fine with this being merged.

**Comment by mhilbrunner** [comment]
Thank you for your work, quite some effort that has gone into this! 🎉 Merged.

**Pull request merged**